### PR TITLE
[PP-25] Emit events on buttons click

### PR DESCRIPTION
**Pull request by andreikaptsiuh**
Task: https://andrei-lysiuk.atlassian.net/browse/PP-25
Description: Discuss with BE what payload should include events and create enum for it
Button events:
- start-game
- restart-game
- finish-game
- user-vote
- create-game

### CODE REVIEW

**Review by Praddby — APPROVED** [review]

File: src/App.tsx [hunk 1]
```diff
@@ -10,23 +10,21 @@ import { Home } from './pages/Home';
...
-          </Route>
-          <Route>
-            <NotFound />
-          </Route>
-        </Switch>
-      </Router>
-    </div>
-  );
-}
+export const App: React.FC = () => (
```
Comment (mrakofff) [inline src/App.tsx]: Any reason for this update?

File: src/interfaces.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 export interface IUser {
   id: string;
-  role?: string;
   name: string;
+  role?: string;
```
Comment (mrakofff) [inline src/interfaces.ts]: any reason for this update?

File: src/mocks/users.json [hunk 3]
```diff
@@ -7,7 +7,8 @@
     "job": "web-developer",
     "avatar": "",
     "currentScore": "",
-    "isCardOpened": false
+    "isCardOpened": false,
```
Comment (mrakofff) [inline src/mocks/users.json]: will we have this field?

File: src/utils/enums/ButtonEvent.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+export enum ButtonEvent {
```
Comment (mrakofff) [inline src/utils/enums/ButtonEvent.ts]: I suppose that these events belong to `SocketEvent`

File: src/components/VoteCard/VoteCard.tsx [hunk 5]
```diff
@@ -29,7 +29,8 @@ export const VoteCard: FC<IVoteCardProps> = ({
 
   function onCardClick(event: MouseEvent): void {
     const currentElem = event.target as HTMLElement;
-    setSelectedCardValue(currentElem.textContent!);
+    if (!currentElem.textContent) return;
+    setSelectedCardValue(currentElem.textContent);
```
Comment (mrakofff) [inline src/components/VoteCard/VoteCard.tsx:35]: I'm not sure that we should use `textContent` coz we could bind click with cardItem data

File: src/components/VoteCard/VoteCard.tsx [hunk 6]
```diff
@@ -29,7 +29,8 @@ export const VoteCard: FC<IVoteCardProps> = ({
 
   function onCardClick(event: MouseEvent): void {
     const currentElem = event.target as HTMLElement;
-    setSelectedCardValue(currentElem.textContent!);
+    if (!currentElem.textContent) return;
```
Comment (mrakofff) [inline src/components/VoteCard/VoteCard.tsx]: don't use 1 line condition

File: src/components/NewGameModal/NewGameModal.tsx [hunk 7]
```diff
@@ -45,7 +47,7 @@ export const NewGameModal: React.FC<INewGameModalProps> = ({ show, onClick }) =>
 
   const submitHandler: FormEventHandler = (event): void => {
     event.preventDefault();
-    console.log('submit', form);
+    socketService.emit(ButtonEvent.CreateGame);
```
Comment (mrakofff) [inline src/components/NewGameModal/NewGameModal.tsx]: Should we add any condition to check if data is filled?

**Review by mrakofff — CHANGES_REQUESTED** [review]

File: src/mocks/users.json (hunk 3 above)
Comment (andreikaptsiuh) [reply src/mocks/users.json]: it's from develop brunch, remove it?

File: src/interfaces.ts (hunk 2 above)
Comment (andreikaptsiuh) [reply src/interfaces.ts]: fix

File: src/App.tsx (hunk 1 above)
Comment (andreikaptsiuh) [reply src/App.tsx]: It's from develop

File: src/utils/enums/ButtonEvent.ts (hunk 4 above)
Comment (andreikaptsiuh) [reply src/utils/enums/ButtonEvent.ts]: fix

File: src/components/VoteCard/VoteCard.tsx (hunk 6 above)
Comment (andreikaptsiuh) [reply src/components/VoteCard/VoteCard.tsx]: fix

File: src/components/NewGameModal/NewGameModal.tsx (hunk 7 above)
Comment (andreikaptsiuh) [reply src/components/NewGameModal/NewGameModal.tsx]: fix

File: src/components/VoteCard/VoteCard.tsx (hunk 5 above)
Comment (andreikaptsiuh) [reply src/components/VoteCard/VoteCard.tsx:35]: This is not my code, it's from develop

**Review by mrakofff — APPROVED** [review]

**Pull request merged**